### PR TITLE
Docs and Scripts to support ARC enabling of DO clusters

### PR DESCRIPTION
**Pull request by hattan**
## Purpose of PR

- Docs and Scripts to support ARC enabling of DO clusters

# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

### CODE REVIEW

File: do-vms/README.md [hunk 1]
```diff
@@ -7,3 +7,93 @@
...
+  - To create an SP(service Principal) (note one was been created for adkc in the tld key vault):
+
+  ```bash
+            az ad sp create-for-rbac\
+            --name "<SP_NAME>"\
+            --skip-assignment\
+            --scope "/subscriptions/<SUBSCRIPTION_ID>"`
+  ```
+
+  - Assign the Owner role to the SP
```
Comment (bartr) [inline do-vms/README.md:29]: can we restrict this role?  owner is a pretty big surface

File: do-vms/README.md [hunk 2]
```diff
@@ -7,3 +7,93 @@
...
+To configure Azure Arc for Digial Ocean Clusters:
+
+- Authentication:
+  Because the VMs running these cluster are on digital ocean, we cannot use an Azure Managed identity. Instead we will have to rely on an Service Principal.
+
+  - To create an SP(service Principal) (note one was been created for adkc in the tld key vault):
+
+  ```bash
+            az ad sp create-for-rbac\
+            --name "<SP_NAME>"\
```
Comment (bartr) [inline do-vms/README.md:24]: use env vars for DevX

File: do-vms/README.md [hunk 3]
```diff
@@ -7,3 +7,93 @@
...
+  AKDC_SP_JSON=$(az keyvault secret show --vault-name "$AKDC_VAULT_NAME" -n "$AKDC_VAULT_KEY" --query "value" -o tsv )
+
+  # get appId
+  echo $AKDC_SP_JSON | jq '.appId'
+
+  # get password
+  echo $AKDC_SP_JSON | jq '.password'
+
+    ```
+
```
Comment (bartr) [inline do-vms/README.md:54]: you need to scp .ssh files as well as vm-setup/akdc (with -r)  

that should copy the individual files

File: do-vms/README.md [hunk 4]
```diff
@@ -7,3 +7,93 @@
...
+  `scp -P 2222 -o "StrictHostKeyChecking=no" arc-setup.sh akdc@<DO VM IP>:~/arc-setup.sh`
+
+- Using scp, copy the get-service-account-token.sh file to the digital ocean cluster. Note this should be the [get-service-account-token.sh](./get-service-account-token.sh) file in the do-vms folder of this repo.
+
+  `scp -P 2222 -o "StrictHostKeyChecking=no" get-service-account-token.sh akdc@<DO VM IP>:~/arc-setup.sh`
+- ssh into the DO cluster
+- NOTE: The following steps should be performed on the DO VM itself:
+  - Run the following commands to add the azure-cli to the package repo list:
+    - `curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null`
```
Comment (bartr) [inline do-vms/README.md:66]: we already add the package in akdc.template

File: do-vms/README.md [hunk 5]
```diff
@@ -7,3 +7,93 @@
...
+    - `echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/azure-cli.list`
+    - `apt-get update`
+  - Configure environment variables for the cluster:
```
Comment (bartr) [inline do-vms/README.md:69]: some of these should be added to /etc/bash.bashrc by the template

File: do-vms/README.md (hunk 2 above)
Comment (hattan) [reply do-vms/README.md:24]: For creating the SP? We don't need to create an SP if it's in the keyvault. I just included the create to provide extra documentation if folks don't have an SP available and to walk them through how to create one.

File: do-vms/README.md (hunk 3 above)
Comment (hattan) [reply do-vms/README.md:54]: Sounds good, we can definitely move to sending the entire folder via scp. I tried to keep it very focused on just the files needed for this scenario, but I like moving it to the entire folder.

File: do-vms/README.md (hunk 4 above)
Comment (hattan) [reply do-vms/README.md:66]: Yes, I just wanted to document this for DO clusters that are hand created also. If using adkc.template with scripts we can remove this .

File: do-vms/README.md (hunk 5 above)
Comment (hattan) [reply do-vms/README.md:69]: Absolutely, if we are creating these DO servers via a script these env vars are added automatically to .bashrc

File: do-vms/README.md (hunk 3 above)
Comment (bartr) [reply do-vms/README.md:54]: these are injected into the SSH keys that DO deploys when it creates the cluster. You have to create the SSH keys with the proper values. Once you do that, it's automatic.

You also have to have the shared SSH key in your DO account.

There are 3 additional ssh keys to create - SP_TENANT, SP_NAME and SP_KEY - they have to be valid and unique SSH keys

ssh-rsa uniqueValidSSHKey  SP_TENANT base64TenantId
ssh-rsa uniqueValidSSHKey  SP_NAME base64SPName
ssh-rsa uniqueValidSSHKey  SP_KEY base64SPKey

during cloud-init, these values get extracted and decoded into ~/.ssh/*

Setup will fail if az login fails.

Here's a key for $100 credit for 30 days to test with:  https://m.do.co/c/30537b10ba7c

LMK if you want to pair and I can walk someone through setup.

@hattan @BSamodien @aflinchb @kev-ms @devwag

File: do-vms/README.md (hunk 4 above)
Comment (bartr) [reply do-vms/README.md:66]: we don't use SCP any longer - we use git clone

Which won't work without az login as that's where our PAT is.

File: do-vms/README.md (hunk 5 above)
Comment (bartr) [reply do-vms/README.md:69]: SP_* is saved to ~/.ssh during cloud-init - assuming az login worked

File: do-vms/README.md (hunk 1 above)
Comment (bartr) [reply do-vms/README.md:29]: there's an issue on the board for this, so I'm resolving for a future PR

File: do-vms/README.md (hunk 2 above)
Comment (bartr) [reply do-vms/README.md:24]: right, but they have to edit the multi-line command. Everywhere else, we set the two env vars on separate lines and then use in the command. A lot easier and consistent.

you can likely replace <SUBSCRIPTION_ID> with $(az account ...)

**Pull request closed without merging**